olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/TileRequest.h

```diff
@@ -0,0 +1,154 @@
+ public:
+  /**
+   * @brief Gets the billing tag to group billing records together.
+   *
+   * The billing tag is an optional free-form tag that is used for grouping
+   * billing records together. If supplied, it must be 4–16 characters
+   * long and contain only alphanumeric ASCII characters [A-Za-z0-9].
+   *
+   * @return The `BillingTag` string or `boost::none` if the billing tag is not
+   * set.
+   */
+  inline const boost::optional<std::string>& GetBillingTag() const {
+    return billing_tag_;
+  }
+
+  /**
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param tag The `BillingTag` string or `boost::none`.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithBillingTag(boost::optional<std::string> tag) {
+    billing_tag_ = std::move(tag);
+    return *this;
+  }
+
+  /**
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param tag The rvalue reference to the `BillingTag` string or
+   * `boost::none`.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithBillingTag(std::string tag) {
+    billing_tag_ = std::move(tag);
+    return *this;
+  }
+
+  /**
+   * @brief Sets the tile key for the request.
+   *
+   * @param tile_key Tile key value.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithTileKey(geo::TileKey tile_key) {
+    tile_key_ = std::move(tile_key);
+    return *this;
+  }
+
+  /**
+   * @brief Gets a tile key value.
+   *
+   * @return The tile key or `boost::none` if tile_key is not set.
+   */
+  inline const geo::TileKey& GetTileKey() const { return tile_key_; }
+
+  /**
+   * @brief Gets the fetch option that controls how requests are handled.
+   *
+   * The default option is `OnlineIfNotFound` that queries the network if
+   * the requested resource is not in the cache.
+   *
+   * @return The fetch option.
+   */
+  inline FetchOptions GetFetchOption() const { return fetch_option_; }
+
+  /**
+   * @brief Sets the fetch option that you can use to set the source from
+   * which data should be fetched.
+   *
+   * @see `GetFetchOption()` for information on usage and format.
+   *
+   * @param fetch_option The `FetchOption` enum.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithFetchOption(FetchOptions fetch_option) {
+    fetch_option_ = fetch_option;
+    return *this;
+  }
+
+  /**
+   * @brief Creates a readable format for the request.
+   *
+   * @param layer_id The ID of the layer that is used for the request.
+   *
+   * @return A string representation of the request.
+   */
+  inline std::string CreateKey(const std::string& layer_id) const {
+    std::stringstream out;
+    out << layer_id << " key:" << GetTileKey();
```

Comment:
E.g. for the DataRequest the key is: `versioned_test_layer[p17626]@100^0`.
I would do it similar for TileRequest: `versioned_test_layer[t17626]@100^0`

With the small `t` and `p` at the beginning of the partition id we can see if it is a **p**artitions or a **t**ile request.